activesupport/lib/active_support/core_ext/object/with.rb

```diff
@@ -28,7 +28,7 @@ def with(**attributes)
         old_values[key] = public_send(key)
         public_send("#{key}=", value)
       end
-      yield
+      yield self
```

Comment:
I took the liberty to improve the implementation though, calling `yield_safe(&block)` instead of just `yield safe` in unnecessary overhead AFAICT.

Reply:
While I acknowledge that the change is a mix of optimization and taste, I wonder if the documentation would benefit from the `&block` argument's presence in the method signature. While the examples outline that there's a block involved, I find it easier for readers to grasp when a block argument is explicitly mentioned.

Reply:
Yeah, it's more the role of the documentation. I can update it further.

Reply:
How is it now?

Reply:
I think that's an improvement!